### PR TITLE
Support automation of the completion tooltip

### PR DESCRIPTION
**Pull request by rchande**
Because completion asynchronously computes descriptions, the default
text is "...". In order to present the description when available, we
have to set the AutomationProperties.Name on our tooltip UIElement.

Note @olegtk: I'm having trouble testing this because the automation focus moves to the text editor as soon as the tooltip comes up.
Tag @olegtk @dotnet/roslyn-ide @dpoeschl for review

**Customer scenario**

Customer is using a screen reader and wants to read the completion tooltip. Currently it is always set to `...`

**Bugs this fixes:**

https://devdiv.visualstudio.com/DevDiv/_workitems?id=433619&fullScreen=true&_a=edit

**Workarounds, if any**

None, there's no way to read the tooltip with a screen reader.

**Risk**

Low. Once we've computed the tooltip we set an automationproperty that the editor reads for us. 

**Performance impact**

None, we're setting a property using data we have to compute anyway.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Didn't test this accessibility scenario.

**How was the bug found?**

Accessibility testing.

### CODE REVIEW

**Comment by dpoeschl** [comment]
Needs a better PR title

**Comment by dpoeschl** [comment]
(And preferably commit title)

**Comment by dpoeschl** [comment]
> I'm having trouble testing this because the automation focus moves to the text editor as soon as the tooltip comes up.

For me, there's about a second where the item is focused. If it's similar for you, you should have enough time to check the HelpText. Can you do that and report back?

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/ToolTipProvider.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@
 using Microsoft.VisualStudio.Text.Classification;
 using Microsoft.VisualStudio.Utilities;
 using VSCompletion = Microsoft.VisualStudio.Language.Intellisense.Completion;
+using System.Windows.Automation;
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/ToolTipProvider.cs]: Sort usings

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/ToolTipProvider.cs [hunk 2]
```diff
@@ -89,6 +90,11 @@ private void ProcessDescription(Task<CompletionDescription> obj)
 
                 var description = obj.Result;
                 this.Content = GetTextBlock(description.TaggedParts, _toolTipProvider._typeMap);
+
+                // As of https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/67974?_a=overview
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/ToolTipProvider.cs]: I personally don't think you need to cite the PR where the editor started doing this. Just explaining that they do it should be sufficient.

**Comment by olegtk** [comment]
It's tricky all right. Start inspect.exe, toggle "Watch Cursor" toolbar button on, invoke completion, arrow down to select a completion item. Hover mouse over it until you see in inspect.exe the "menu item", check its Help Text property.

**Comment by rchande** [comment]
Thanks @olegtk. It doesn't look like the editor is pulling in the Name we set. I'm setting it on the actual UIELement returned by `GetUIElement`. Is that the right place? Does the editor cache this value (and therefore not pick up the updated name?

I'm on d15prerel 26518.0.

**Comment by olegtk** [comment]
Hmmm, let me check that.

**Comment by rchande** [comment]
Ping @olegtk ?

**Comment by olegtk** [comment]
I verified it on a simple extension and it just works. I guess the problem is that we grab Name property value immediately after you return this object and in your case I guess it's async (initially there is no Name value and only when it's calculated you set it, right?).
If that's true, we need to listen to Name property changes. Let me verify that would work tomorrow.

**Comment by rchande** [comment]
@olegtk Right--we're updating the Name property once we have a value, so you'll have to refresh your cache.

**Comment by olegtk** [comment]
let me verify just changing Name property will work (it should as it's a dependency property) before you complete this PR and send the bug back to me.

**Comment by rchande** [comment]
retest windows_release_vs-integration_prtest

**Comment by rchande** [comment]
Oleg and I confirmed offline that this should work and we should merge so it can be debugged from the editor side.

**Comment by rchande** [comment]
Tag @MattGertz for ask mode approval

**Comment by dpoeschl** [comment]
@dotnet/roslyn-infrastructure This has been in the queues for 22 hours and is still 93/149, 16/39, 90/149, 91/149, 92/149 in the queues. Is something wrong?

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest this please.

**Comment by jasonmalinowski** [comment]
@dpoeschl: Jenkins was restarted, which means anything in the queue gets blown away and needs to be retested.

**Comment by dpoeschl** [comment]
retest windows_release_vs-integration_prtest please

**Comment by dpoeschl** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by dpoeschl** [comment]
retest windows_release_vs-integration_prtest

**Comment by dpoeschl** [comment]
retest windows_release_vs-integration_prtest

**Pull request merged**